### PR TITLE
Renamed `PerDefTables` to `Tables`

### PR DESCRIPTION
**Pull request by Lezzz**
Originally suggested by @eddyb

### CODE REVIEW

File: src/librustc_metadata/rmeta/encoder.rs [hunk 1]
```diff
@@ -585,7 +585,7 @@ impl<'tcx> EncodeContext<'tcx> {
             println!("  def-path table bytes: {}", def_path_table_bytes);
             println!(" proc-macro-data-bytes: {}", proc_macro_data_bytes);
             println!("            item bytes: {}", item_bytes);
-            println!("   per-def table bytes: {}", per_def_bytes);
+            println!("   per-def table bytes: {}", tables_bytes);
```
Comment (eddyb) [inline src/librustc_metadata/rmeta/encoder.rs]: "per-def table bytes" could probably just say "table bytes".

File: src/librustc_metadata/rmeta/encoder.rs [hunk 2]
```diff
@@ -585,7 +585,7 @@ impl<'tcx> EncodeContext<'tcx> {
             println!("  def-path table bytes: {}", def_path_table_bytes);
```
Comment (eddyb) [inline src/librustc_metadata/rmeta/encoder.rs]: Not relevant to this PR necessarily but we should probably encode the `DefPathTable` as an actual `Table`.

Comment (eddyb) [reply src/librustc_metadata/rmeta/encoder.rs]: Harder to do without splitting `DefPathTable` into several tables, though.

**Comment by oli-obk** [comment]
@bors r+

This was much more self-contained than I would have thought from the title.

**Pull request merged**